packages/web3-eth/src/schemas.ts

```diff
@@ -321,13 +321,37 @@ export const blockSchema = {
 	},
 };
 
+export const withdrawalsSchema = {
+	type: 'object',
+	properties: {
+		index: {
+			format: 'uint',
+		},
+		validatorIndex: {
+			format: 'uint',
+		},
+		address: {
+			format: 'bytes32',
+		},
+		amount: {
+			format: 'uint',
+		},
+	},
+};
+
 export const blockHeaderSchema = {
 	type: 'object',
 	properties: {
+		author: {
```

Comment:
some of these fields are not aligned with EL specs, there can be possibilities:

- EL specs not updated
- geth specific fields

so could you add an integration test , so when we add more EL clients we can catch issue earlier.

Reply:
@jdevcs What EL spec are you looking at? I've been using [this](https://ethereum.github.io/execution-apis/api-documentation/) one and don't even see `eth_subscribe` listed

Regardless, these added fields don't exist on Geth's response, but are instead from Nethermind's. It's okay and actually necessary to have these fields in the `blockHeaderSchema`, because it's used to format responses. Meaning as is, any fields not mentioned in this schema will be dropped from the response before it's returned to the user. No error is caused if the fields are not present. This does, however, beg the question of how do we declare the [BlockHeaderOutput](https://github.com/web3/web3.js/blob/6992ccdd555604ff22d92c0eba0aa5f366f8b3cf/packages/web3-types/src/eth_types.ts#L154) type - do we include these properties as optional since they _could_ be there if the connected RPC client uses them? Or do we only include the common ones as the type is declared now?

The extra fields included by Nethermind that don't seem to be available using Geth:

- author
- totalDifficulty
- size
- excessDataGas
- mixHash
- transactions
- uncles
- withdrawals

Reply:
I wasnt able to find `author` field in EL Specs https://github.com/ethereum/execution-apis/blob/4da6a02f094d9d8c826272f6ceac2d751866f132/src/schemas/block.yaml#L22 in block schema at first place, and later there is no `eth_subscribe` in EL specs ( as you found as well in specs doc). 


If we include integration tests ( we will need to have a check for Nethermind in tests and expect that these additional fields are present), if its breaking in future for Nethermind we will be able to detect. Currently we dnt have NM, so it can be skipped.

> This does, however, beg the question of how do we declare the [BlockHeaderOutput](https://github.com/web3/web3.js/blob/6992ccdd555604ff22d92c0eba0aa5f366f8b3cf/packages/web3-types/src/eth_types.ts#L154) type - do we include these properties as optional since they could be there if the connected RPC client uses them? Or do we only include the common ones as the type is declared now?

I'll suggest lets have additional types as optional and add doc comments that these are specific for NM.